### PR TITLE
Refactored hook handlers in mongoose_metrics module

### PR DESCRIPTION
**Pull request by pawlooss1**
This PR changes all hook handlers in `mongoose_metrics` module to `gen_hook` format.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/143878) / small_tests / 25cb42df0f1291374c42a012382d7789331341b6
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3808/143878/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3808/143878/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-10-12_11.09.23/lib.mongooseim.logs/run.2022-10-12_11.09.23/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/144298) / small_tests / b6ec3e364bb471bc386069da8d3171e88259d155
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3808/144298/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3808/144298/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-10-13_08.06.11/lib.mongooseim.logs/run.2022-10-13_08.06.12/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/144325) / small_tests / 34ae1784d2dd9da1305770d535941888bbbb465c
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3808/144325/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3808/144325/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-10-13_08.08.58/lib.mongooseim.logs/run.2022-10-13_08.08.59/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/144447) / small_tests / 4d4391e51834ec107b2fe812e6bad4fb900989fc
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3808/144447/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3808/144447/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-10-13_08.50.48/lib.mongooseim.logs/run.2022-10-13_08.50.48/suite.log.html)

File: src/metrics/mongoose_metrics_hooks.erl [hunk 1]
```diff
@@ -16,89 +16,91 @@
...
--spec sm_remove_connection_hook(mongoose_acc:t(), tuple(), jid:jid(),
-                                term(), ejabberd_sm:close_reason()
-                               ) -> mongoose_acc:t().
-sm_remove_connection_hook(Acc, _, _, _, _Reason) ->
+-spec sm_remove_connection_hook(Acc, Params, Extra) -> {ok, Acc} when
+      Acc :: mongoose_acc:t(),
+      Params :: map(),
+      Extra :: map().
+sm_remove_connection_hook(Acc, _, _) ->
     HT = mongoose_acc:host_type(Acc),
```
Comment (NelsonVides) [inline src/metrics/mongoose_metrics_hooks.erl:79]: Actually, just like for `sm_register_connection_hook` we're extracting the `host_type` from the `Extra` argument, we can do the same in this one and the many other hooks that extract it ad-hoc from the `mongoose_acc`.

**Review by NelsonVides — COMMENTED** [review]
Just one comment about how to extract host types 🙃

File: src/metrics/mongoose_metrics_hooks.erl (hunk 1 above)
Comment (pawlooss1) [reply src/metrics/mongoose_metrics_hooks.erl]: done 👍

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/144721) / small_tests / a2328acce550868deef129bbb3b8607d8af201c0
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3808/144721/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3808/144721/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-10-13_13.49.50/lib.mongooseim.logs/run.2022-10-13_13.49.50/suite.log.html)

**Review by NelsonVides — APPROVED** [review]
LGTM 👌🏽

**Pull request merged**